### PR TITLE
Add support for sending output to something besides stdout

### PR DESCRIPTION
**Pull request by jlucier**
Added versatility to the `stdout` parameter of the profile hook / profile classes.

Specifying a "file-like" object will send all the output of `FuncProfile.print_stats` to that file object instead of `sys.stdout`. Feedback welcome.

### CODE REVIEW

File: profilehooks.py
Comment (mgedmin) [inline profilehooks.py:429]: Curious: why do it this way instead of making self.print_stats() do `print(..., file=stream`) a few times?

AFAIU self.stats.print_stats() already prints to `stream`?

**Review by mgedmin — APPROVED** [review]
Thank you, this looks nice!

Would you like to write a changelog note for this feature?

(Apologies for a late reply, I was on vacation and was ignoring computers.)

File: profilehooks.py
Comment (jlucier) [reply profilehooks.py:429]: I completely overlooked the `print(..., file=stream)` option, and you're right that `self.stats.print_stats()` will already go to the right place.

I'll make a quick port over to that method, it's simpler. I think I have a way to avoid repeating `file=stream` a bunch so as to make it hard for someone to change / add prints in `print_stats` and forget that kwarg.

**Comment by jlucier** [comment]
Happy to write a changelog, should I add it directly to `CHANGES.rst`?

**Comment by mgedmin** [comment]
> Happy to write a changelog, should I add it directly to CHANGES.rst?

Yes, please!

**Review by mgedmin — APPROVED** [review]
LGTM!  I'll wait for the promised changelog bit before hitting the merge button ;)

Thanks!

File: profilehooks.py
Comment (mgedmin) [inline profilehooks.py:382]: FWIW if you're curious why I had `print("")` here instead of a simple `print()`, I had this silly habit of writing code compatible with Python 2 and 3 at the same time without relying on `from __future__ import print_function`.  `print("")` works fine on both Pythons, `print()` without the future import prints the representation of an empty tuple on Python 2.

Anyway, the future import was added a long time ago and I forgot to remove the now-unnecessary `""`s.  They can stay, they don't bother me too much.

**Comment by mgedmin** [comment]
Released as profilehooks 1.12.0.

**Pull request merged**
